api/CanMsg.h

```diff
@@ -31,14 +33,19 @@ namespace arduino
 class CanMsg : public Printable
 {
 public:
-  static size_t constexpr MAX_DATA_LENGTH = 8;
+  static uint8_t  constexpr MAX_DATA_LENGTH = 8;
```

Comment:
Of course, the thing that is most irksome when you finally add CAN-FD support is the quantization of the DLC. If you are trying to get ahead of adoption and avoid future churn you might want to start moving away from an integer data_length type to an enumerated one now.

Reply:
I had been thinking of making a `CanFdMsg` whose `MAX_DATA_LENGTH` would be 64 then. Overall I'm a fan of `const`/`constexpr` over `enum` because they provide a strong type guarantee (vs. `enum` that easily and without warning converts to int, etc.).

_Note_: enum classes in public Arduino APIs are discouraged (they'd provide strong type support).